lib/rest2.js

```diff
@@ -1229,10 +1231,12 @@ class RESTv2 {
    * @param {Function} [cb] - callback
    * @returns {Promise} p
    */
-  generateToken ({ ttl, scope, caps, writePermission, _cust_ip: custIp } = {}, cb) {
-    return this._makeAuthRequest('/auth/w/token', {
-      ttl, scope, caps, writePermission, _cust_ip: custIp
-    }, cb)
+  generateToken (opts, cb) {
+    let params = _pick(opts || {}, ['ttl', 'scope', 'caps', 'writePermission', '_cust_ip'])
+    params = _pickBy(params, Boolean) // clear falsy values
```

Comment:
this does not work with writePermission, which is a boolean:

```js
  const response = await rest2.generateToken({
    scope: 'blerg',
    writePermission: false
  })
```

Reply:
adjusted the pickBy filter to check nil values